addons/mail/static/src/core/common/chat_hub_model.js

```diff
@@ -48,53 +50,40 @@ export class ChatHub extends Record {
     });
     /** From top to bottom. Bottom-most will actually be hidden */
     folded = Record.many("ChatWindow", { inverse: "hubAsFolded" });
+    initPromise = new Deferred();
     loadMutex = new Mutex();
 
-    closeAll() {
+    async closeAll() {
+        await this.initPromise;
+        const promises = [];
         for (const cw of [...this.opened, ...this.folded]) {
-            cw.close({ notifyState: false });
+            promises.push(cw.close({ notifyState: false }));
         }
+        await Promise.all(promises);
         this.save(); // sync only once at the end
     }
 
-    init() {
-        const { opened = [], folded = [] } = JSON.parse(
-            browser.localStorage.getItem(CHAT_HUB_KEY) ?? "{}"
-        );
-        for (const threadData of [...opened, ...folded]) {
-            this.initThread(threadData);
-        }
-    }
-
-    initThread(threadData) {
-        this.store.fetchStoreData("mail.thread", {
-            thread_model: threadData.model,
-            thread_id: threadData.id,
-            request_list: ["display_name"],
-        });
-    }
```

Comment:
Note: now fully relying on `getOrFetch`.
All of this was only necessary to prefetch or compensate `getOrFetch` not actually fetching threads.